File: Course_01/Java Language/Matrizes/matriz_geral.java
package Matrizes;

import java.util.Locale;
import java.util.Scanner;

public class matriz_geral {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		
		Scanner sc = new Scanner(System.in);
		
		int i, j, m, linha, coluna;
		
		System.out.print("Qual a ordem da matriz? ");
		m = sc.nextInt();
		
		double [][] matriz = new double [m][m];
		
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				matriz[i][j] = sc.nextDouble();
			}
		}
		
		double soma;
		
		soma = 0;
		
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				if (matriz[i][j] > 0) { 
					soma += matriz[i][j];
				}
			}
		}
		
		System.out.println();
		System.out.println("SOMA DOS POSITIVOS: " + soma);
		System.out.println();
		
		System.out.print("Escolha uma linha: ");
		linha = sc.nextInt();
		System.out.print("LINHA ESCOLHIDA: ");
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				if (i == linha) {
					System.out.print(matriz[i][j] + " ");
				}
			}
		}
		System.out.println();
		System.out.println();
		System.out.print("Escolha uma coluna: ");
		coluna = sc.nextInt();
		System.out.print("COLUNA ESCOLHIDA: ");
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				if (j == coluna) {
					System.out.print(matriz[i][j] + " ");
				}
			}
		}
		System.out.println();
		System.out.println();
		System.out.print("DIAGONAL PRINCIPAL: ");
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				if (i == j) {
					System.out.print(matriz[i][j] + " ");
				}
			}
		}
		System.out.println();
		System.out.println();
		
		double [][] matrizAlterada = new double [m][m];
		
		System.out.println("MATRIZ ALTERADA:");
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				if (matriz[i][j] < 0) {
					matrizAlterada[i][j] = matriz[i][j] * matriz[i][j];
				}
				else {
					matrizAlterada[i][j] = matriz[i][j];	
				}
				System.out.print(matrizAlterada[i][j] + " ");
			}
			System.out.println();
		}
		sc.close();
	}

}

File: Course_01/Python/Matrizes/matriz_geral.py
m = int(input("Qual a ordem da matriz? "))
n = m

matriz = [[0 for x in range(n)] for x in range(m)]

for i in range(0, m):
    for j in range(0, n):
        matriz[i][j] = float(input(f"Elemento [{i},{j}]: "))

somaPositivos = 0
for i in range(0, m):
    for j in range(0, n):
        if matriz[i][j] > 0:
            somaPositivos += matriz[i][j]
print()
print(f"Soma dos positivos: {somaPositivos:.1f}")
print()
linha = int(input("Escolha uma linha:"))
print(f"Linha escolhida: ", end="")
for j in range(0, n):
    print(f"{matriz[linha][j]:.1f} ", end="")
print()
print()
coluna = int(input("Escolha uma coluna:"))
print(f"Coluna escolhida: ", end="")
for i in range(0, n):
    print(f"{matriz[i][coluna]:.1f} ", end="")
print()
print()
print(f"Diagonal principal: ", end="")
for i in range(0, n):
    print(f"{matriz[i][i]:.1f} ", end="")
print()
print()
print("Matriz Alterada:")
matrizAlterada = [[0 for x in range(n)] for x in range(m)]
for i in range(0, m):
    for j in range(0, n):
        if matriz[i][j] < 0:
            matrizAlterada[i][j] = matriz[i][j] ** 2
        else:
            matrizAlterada[i][j] = matriz[i][j]
for i in range(0, m):
    for j in range(0, n):
        print(f"{matrizAlterada[i][j]:.1f} ", end="")
    print()